sdk/deepbook/src/deepbook_sdk.ts

```diff
@@ -176,99 +219,152 @@ export class DeepBook_sdk {
-	 * @param amountIn: amount of token to buy or sell, eg: 10000000
-	 * @param currentAddress: current user address, eg: "0xbddc9d4961b46a130c2e1f38585bbc6fa8077ce54bcb206b26874ac08d607966"
+	 * @param tokenObjectIn Object id of the token to swap: eg: "0x6e566fec4c388eeb78a7dab832c9f0212eb2ac7e8699500e203def5b41b9c70d"
+	 * @param amountIn amount of token to buy or sell, eg: 10000000
+	 * @param currentAddress current user address, eg: "0xbddc9d4961b46a130c2e1f38585bbc6fa8077ce54bcb206b26874ac08d607966"
+	 * @param accountCap Object id of Account Capacity under user address, created after invoking createAccount, eg: "0x6f699fef193723277559c8f499ca3706121a65ac96d273151b8e52deb29135d3"
 	 */
 	public swap_exact_base_for_quote(
 		token1: string,
 		token2: string,
+		client_order_id: string,
 		poolId: string,
-		treasury: string,
 		tokenObjectIn: string,
 		amountIn: number,
 		currentAddress: string,
+		accountCap: string,
 	): TransactionBlock {
 		const txb = new TransactionBlock();
 		// in this case, we assume that the tokenIn--tokenOut always exists.
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
 		const [base_coin_ret, quote_coin_ret, _amount] = txb.moveCall({
 			typeArguments: [token1, token2],
-			target: `dee9::clob::swap_exact_base_for_quote`,
+			target: `${PACKAGE_ID}::${MODULE_CLOB}::swap_exact_base_for_quote`,
 			arguments: [
-				txb.object(String(poolId)),
+				txb.object(poolId),
+				txb.pure(client_order_id),
+				txb.object(accountCap),
 				txb.object(String(amountIn)),
 				txb.object(tokenObjectIn),
-				// we have to mint a zero amount of token2 in the treasury, only with this object can we pass it into the function.
-				this.mint(token2, 0, treasury, txb),
-				txb.object(normalizeSuiObjectId('0x6')),
+				txb.moveCall({
+					typeArguments: [token2],
+					target: `0x2::coin::zero`,
+					arguments: [],
+				}),
+				txb.object(SUI_CLOCK_OBJECT_ID),
 			],
 		});
 		txb.transferObjects([base_coin_ret], txb.pure(currentAddress));
 		txb.transferObjects([quote_coin_ret], txb.pure(currentAddress));
-		txb.setSenderIfNotSet(currentAddress);
-		txb.setGasBudget(this.gasBudget);
 		return txb;
 	}
 
 	/**
 	 * @description: place a limit order
 	 * @param token1 Full coin type of the base asset, eg: "0x3d0d0ce17dcd3b40c2d839d96ce66871ffb40e1154a8dd99af72292b3d10d7fc::wbtc::WBTC"
 	 * @param token2 Full coin type of quote asset, eg: "0x3d0d0ce17dcd3b40c2d839d96ce66871ffb40e1154a8dd99af72292b3d10d7fc::usdt::USDT"
+	 * @param client_order_id
 	 * @param poolId Object id of pool, created after invoking createPool, eg: "0xcaee8e1c046b58e55196105f1436a2337dcaa0c340a7a8c8baf65e4afb8823a4"
 	 * @param price: price of the limit order, eg: 180000000
 	 * @param quantity: quantity of the limit order in BASE ASSET, eg: 100000000
+	 * @param self_matching_prevention: true for self matching prevention, false for not, eg: true
```

Comment:
For ergonomics of people using this function, it might also be worth defaulting this to true.

Reply:
At the moment only `CANCEL_OLDEST` has been implemented https://github.com/MystenLabs/sui/commit/652fa40efe5c90bf96a37998d6d59468ed93ddc3

and it does not allow for any other value other than `CANCEL_OLDEST`

https://github.com/MystenLabs/sui/blob/a2a6d6e0d45cca174cbb490b7691ab6ff379fd77/crates/sui-framework/packages/deepbook/sources/clob_v2.move#L1113C17-L1113C41

I'll make it default to 0